pkg/actions/lua/lakefs/catalogexport/delta_exporter.lua

```diff
@@ -193,14 +202,67 @@ local function export_delta_log(action, table_def_names, write_object, delta_cli
             table_physical_path = path_transformer(table_physical_path)
         end
         local table_val = {
-            path=table_physical_path,
-            metadata=metadata,
+            path = table_physical_path,
+            metadata = metadata,
         }
         response[table_name_yaml] = table_val
     end
     return response
 end
 
+-- Function to extract directory from a path
+local function extractDirectory(path)
+    local patt = regexp.compile("^(.*[\\/])")
+    local m = patt.find(path)
+    if m then
+        return m
+    else
+        print("path not found")
+    end
+end
+
+-- Local function to filter the list of table defs to include only those that have changed
+local function changed_table_defs(table_def_names, table_descriptors_path, repository_id, ref, compare_ref)
+    -- Perform a diff_refs operation to get the differences between references
+    local code, diff_resp = lakefs.diff_refs(repository_id, ref, compare_ref)
+    if code ~= 200 then
+        error("Failed to perform diff_refs with code: " .. tostring(code))
+    end
+
+    -- Now make a set out of the paths of the filenames
+    local changed_path_set = {}
+    for index, diff_item in ipairs(diff_resp.results) do
+        local dir = extractDirectory(diff_item.path)
+        if dir then
+            changed_path_set[dir] = true
+            print("  added to changed_path_set:", dir)
+        end
+    end
+
+    -- Initialize the result table for storing changed table definitions
+    local changed_table_def_names = {}
+
+    -- Iterate through the table definitions and add to the result the ones that pass the filter
+    for index, table_name_yaml in ipairs(table_def_names) do
+        local table_descriptor = get_table_descriptor(repository_id, compare_ref, table_name_yaml, table_descriptors_path)
+        print(index, "table_descriptor.path", table_descriptor.path)
+        local table_path = table_descriptor.path .. "/"
+        if not table_path then
+            error("table path is required to proceed with Delta catalog export")
+        end
```

Comment:
You should check if `table_descriptor.path` is nil before you try to concatenate it to a string. If it's nil, it will throw an error when you try to concatenate it.